From 79fbd388d3f66af3cac3fa4f5a4110aa8d0c8a20 Mon Sep 17 00:00:00 2001
From: Xiao Ya peng <yp.xiao@rock-chips.com>
Date: Tue, 10 Dec 2019 09:10:07 +0800
Subject: [PATCH] ARM: dts: rk3229-evb: Enable the integrated PHY for gmac

This patch enables the integrated PHY for rk3229 evb board
by default.
To use the external 1000M PHY on evb board, need to make
some switch of evb board to be on.

Change-Id: I31f2ca6293eeac8a70aca20160f325abdf7509b6
Signed-off-by: Xiao Ya peng <yp.xiao@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-evb-android.dtsi | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3229-evb-android.dtsi b/arch/arm/boot/dts/rk3229-evb-android.dtsi
index 3300d89397d1..965d2ec8b231 100644
--- a/arch/arm/boot/dts/rk3229-evb-android.dtsi
+++ b/arch/arm/boot/dts/rk3229-evb-android.dtsi
@@ -236,8 +236,22 @@
 	clock_in_out = "output";
 	phy-supply = <&vcc_phy>;
 	phy-mode = "rmii";
-	phy-is-integrated;
+	phy-handle = <&phy>;
 	status = "okay";
+
+	mdio {
+		compatible = "snps,dwmac-mdio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		phy: phy@0 {
+			compatible = "ethernet-phy-id1234.d400", "ethernet-phy-ieee802.3-c22";
+			reg = <0>;
+			clocks = <&cru SCLK_MAC_PHY>;
+			resets = <&cru SRST_MACPHY>;
+			phy-is-integrated;
+		};
+	};
 };
 
 &gpu {
-- 
2.35.3

